modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/display/context/TranslateDisplayContext.java

```diff
@@ -352,18 +349,16 @@ public Locale getSourceLocale() {
 	public List<String> getSourceStringValues(
 		InfoField infoField, Locale locale) {
 
-		Collection<InfoFieldValue<Object>> infoFieldValues =
-			_sourceInfoItemFieldValues.getInfoFieldValues(
-				infoField.getUniqueId());
+		List<String> values = new ArrayList<>();
+
+		for (InfoFieldValue<Object> infoFieldValue :
```

Comment:
@JanisZhang Could you review the for-loops and see if they are suitable for using `TransformUtil` or `ListUtil`?